### ISSUE TITLE
'Routing' page issue

### ISSUE THREAD

**Issue by markrchristine** [dart-lang/site-webdev#1833]
From URL: https://webdev.dartlang.org/angular/tutorial/toh-pt5

When i get to "Make the router available" section, you show a major change to the file "main.dart", yet you fail to explain any of it. 
What does this line do?:

`import 'main.template.dart' as self;`

What does this line do?

`final InjectorFactory injector = self.injector$Injector;`

**Comment by kwalrath** [comment]
@chalin do you have time to comment on this? (If something needs changing, I'll be happy to do it. Thanks!)

**Comment by chalin** [comment]
You are correct, we don't cover dependency injection (DI) details in the tutorial. A link to the [Root injector providers](https://webdev.dartlang.org/angular/guide/dependency-injection#root-injector-providers) section would probably be a good idea. WDYT?

(I need to focus on dash this week, so I won't be able to address this until some time next week. If you'd like to add a note+link @kwalrath, please go ahead. Otherwise I can look into it next week. Oh, if you do update, don't refresh site-shared quite yet.)

**Comment by markrchristine** [comment]
This link is perfect, it’s exactly what I needed to make sense of this issue. Thanks.

**Comment by markrchristine** [comment]
Actually, this link answers one question, but another is left unanswered. What is this :
import 'main.template.dart' as self;
I don’t see any explanations in the documentation for this line of code. Is there another link that explains this?

**Comment by Ariesdesire** [comment]
Im currently trying to set this up following the documentation  and when I reach this point I get the following error message
error: The getter 'injector$Injector' isn't defined for the class 'self'. (undefined_getter at [toastmasters_web] web\main.dart:11)

**Comment by KingDarBoja** [comment]
I think this [Medium Article](https://hackernoon.com/router-practical-approach-with-angular-dart-80485978a124) would help to clear your mind regarding your question.

Cheers!
